docs/source/developers/python.rst

```diff
@@ -636,6 +636,52 @@ Caveats
 
 The Plasma component is not supported on Windows.
 
+Deleting stale build artifacts
+==============================
+
+When there have been changes to the structure of the Arrow C++ library or Pyarrow,
+a thorough cleaning is recommended as a first attempt to fixing build errors.
+
+.. note::
+
+   It is not necessarily intuitive from the error itself that the problem is due to stale artifacts.
+   Example of a build error from stale artifacts is "Unknown CMake command "arrow_keep_backward_compatibility".
+
+To delete stale Arrow C++ build artifacts:
+
+.. code-block::
+
+  $ rm -rf arrow/cpp/build
```

Comment:
```suggestion
   $ rm -rf arrow/cpp/build
```

Reply:
@anjakefala to clarify: in practice it doesn't matter how many spaces you are using here (the above also works perfectly fine), but the rest of this file is consistently using 3 spaces for the initial indentation in code blocks